### ISSUE TITLE
Dependabot alert: Command Injection due to insecure usage of the --upload-pack feature of git

### ISSUE THREAD

**Issue by anthonyhaffey** [jaz303/git-clone#15]
Hi there, is there going to be an update to git-clone to deal with the following dependabot alert:

Affected versions <= 0.2.0

**All versions of package git-clone are vulnerable to Command Injection due to insecure usage of the --upload-pack feature of git.**

Side-note: my understanding is that this would be less of a concern for people using node-js for software on their computer, but a bigger issue for node-js on a server, is that a fair assessment?

**Comment by anthonyhaffey** [comment]
Thanks @jaz303 for the quick reply. Sorry that my posting of this issue reflects a lack of understanding about the validity of the dependabot alert; it's a shame that this is being alerted when sensible usage of git-clone isn't a liability (at least based on my understanding of your reply).

I'm happy for you to close/delete this issue.

**Comment by Kristories** [comment]
@jaz303 please check : 
- https://gist.github.com/lirantal/9441f3a1212728476f7a6caa4acb2ccc
- https://security.snyk.io/vuln/SNYK-JS-GITCLONE-2434308
- https://github.com/advisories/GHSA-8jmw-wjr8-2x66

**Comment by jaz303** [comment]
I've read all of these.

`git-clone` doesn't use `upload-pack`; it's explicitly documented as a shell command wrapper, and exposes a method for passing arbitrary arguments to the `git` executable. Obviously care needs to be taken if passing dynamic/untrusted input.  I fail to see how this is a vulnerability any more than a standard library function like `spawn`.

**Comment by Kristories** [comment]
Yes, I know. Perhaps you could provide this information in the readme, as [git-promise](https://github.com/piuccio/git-promise) did.

**Comment by jaz303** [comment]
I did:

> NOTE: the args option allows arbitrary arguments to be passed to git; this is inherently insecure if used in combination with untrusted input. Only use the args option with static/trusted input!

**Comment by Kristories** [comment]
Thanks for the information. I didn't see that warning, because I was too focused on `dependabot` and `npm audit`.

**Comment by emorneau** [comment]
Will we eventually remove that option out of this repository to remove that potential vulnerability. It shows as a Snyk High vuln. which is very annoying even if we do not use this option. Maybe create a new repo git-clone-unload-pack which adds in that extra feature, if people desire to use this risky option.

**Comment by jaz303** [comment]
No I will not be doing that. It's not a "risky option". `git-clone` doesn't directly use `upload-pack`. Did you even read this thread?

**Comment by emorneau** [comment]
I see that it is an inherited security vuln. from git-promise. Sorry and thanks for the quick reply above.

**Comment by hakt0r** [comment]
Honestly why not create a bug with git itself?! It seems to me that some downstream users allow unfiltered arguments to be injected and they are just pointing fingers. This is **not a security issue**, indeed.
